Sources/MapboxMaps/Foundation/MapboxMap.swift

```diff
@@ -835,6 +841,59 @@ extension MapboxMap {
                                  featureId: featureId,
                                  stateKey: stateKey)
     }
+
+}
+
+// MARK: - View Annotations
+
+extension MapboxMap {
+
+    // TODO: Add documentation
+    public func setViewAnnotationPositionsUpdateListenerFor(listener: ViewAnnotationPositionsListener) {
+        __map.setViewAnnotationPositionsUpdateListenerFor(listener)
+    }
+
+    /**
+     * Add view annotation.
+     *
+     * @return position for all views that need to be updated on the screen or null if views' placement remained the same.
+     */
+    public func addViewAnnotation(forId id: String, options: ViewAnnotationOptions) {
+        __map.addViewAnnotation(forIdentifier: id, options: MapboxCoreMaps.ViewAnnotationOptions(options))
+    }
+
+    /**
+     * Update view annotation if it exists.
+     *
+     * @return position for all views that need to be updated on the screen or null if views' placement remained the same.
+     */
+    public func updateViewAnnotation(forId id: String, options: ViewAnnotationOptions) {
+        __map.updateViewAnnotation(forIdentifier: id, options: MapboxCoreMaps.ViewAnnotationOptions(options))
+    }
+
+    /**
+     * Remove view annotation if it exists.
+     *
+     * @return position for all views that need to be updated on the screen or null if views' placement remained the same.
+     */
+    public func removeViewAnnotation(forId id: String) {
+        __map.removeViewAnnotation(forIdentifier: id)
+    }
+
+    // TODO: Add documentation
+    // Make this throw, return ViewAnnotationOptions directly
+    public func getViewAnnotationOptions(forId id: String) throws -> ViewAnnotationOptions {
+        // TODO: error handling
+        let result = __map.getViewAnnotationOptions(forIdentifier: id)
+        guard !result.isError() else {
+            fatalError()
+        }
+        guard let options = result.value as? MapboxCoreMaps.ViewAnnotationOptions else {
+            fatalError()
+        }
+        return ViewAnnotationOptions(options)
+    }
```

Comment:
Are we 100% sure we want to make all of these public?

Reply:
No, I've just forgot to change them to internal.